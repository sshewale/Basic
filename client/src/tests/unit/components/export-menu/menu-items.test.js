import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MenuItems from '../../../../components/export-menu/menu-items';

configure({ adapter: new Adapter() });

const menuItemsProps = {
  items: ['One', 'Two', 'Three'],
  onExport: jest.fn(),
};

describe('MenuItems component', () => {
  let exportItemWrapper;
  beforeAll(() => {
    exportItemWrapper = shallow(<MenuItems {...menuItemsProps} />);
  });
  it('renders correct text in menuItems', () => {
    expect.assertions(4);
    expect(exportItemWrapper.find('.menu-items').get(0).props.children).toEqual(
      'One',
    );
    expect(exportItemWrapper.find('.menu-items').get(1).props.children).toEqual(
      'Two',
    );
    expect(exportItemWrapper.find('.menu-items').get(2).props.children).toEqual(
      'Three',
    );
    expect(exportItemWrapper.find('.menu-items')).toHaveLength(
      menuItemsProps.items.length,
    );
  });
  it('check props of component', () => {
    expect.assertions(2);
    expect(exportItemWrapper.instance().props.items).toBe(menuItemsProps.items);
    expect(exportItemWrapper.instance().props.onExport).toBe(
      menuItemsProps.onExport,
    );
  });
  it('will check the values and props of component', () => {
    expect.assertions(1);
    expect(exportItemWrapper).toMatchSnapshot();
  });
});

describe('MenuItems full dom rendering', () => {
  let exportItemWrapper;
  beforeAll(() => {
    exportItemWrapper = mount(<MenuItems {...menuItemsProps} />);
  });
  afterAll(() => {
    exportItemWrapper.unmount();
  });
  it('check item prop of component', () => {
    expect.assertions(3);
    expect(exportItemWrapper.props().items).toEqual(['One', 'Two', 'Three']);
    expect(exportItemWrapper.props().items).not.toEqual(['abc', 'xyz']);
    expect(exportItemWrapper.props().items).toHaveLength(
      menuItemsProps.items.length,
    );
  });
  it('check setting new prop value to item of component', () => {
    exportItemWrapper.setProps({ items: ['xyz'] });
    expect.assertions(2);
    expect(exportItemWrapper.props().items).toEqual(['xyz']);
    expect(exportItemWrapper.props().items).not.toEqual(['abc']);
  });
});
//TODO test case for onExport props
/* onExport={(item) =>
              this.handleExportMenu(item, refData, exportFileName)
            }
 */
