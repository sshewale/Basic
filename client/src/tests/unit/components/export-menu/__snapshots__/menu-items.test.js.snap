// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItems component will check the values and props of component 1`] = `
<Fragment>
  <WithStyles(ForwardRef(MenuItem))
    className="menu-items"
    key="One"
    onClick={[Function]}
  >
    One
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    className="menu-items"
    key="Two"
    onClick={[Function]}
  >
    Two
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    className="menu-items"
    key="Three"
    onClick={[Function]}
  >
    Three
  </WithStyles(ForwardRef(MenuItem))>
</Fragment>
`;
